lib/system/generic/mbox.h

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (c) 2025 Advanced Micro Devices, Inc. All Rights Reserved.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+/*
+ * @file	mbox.h
+ * @brief	Generic mbox primitives for libmetal.
+ *
+ * Note that for now these are just stubs.
+ */
+
+#ifndef __METAL_MBOX_GENERIC__H__
+#define __METAL_MBOX_GENERIC__H__
+
+#include <metal/sys.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef struct { } metal_mbox_t;
+typedef struct { } metal_mbox_init_data_t;
+typedef void (*metal_mbox_callback_t)(void);
+
+/** \defgroup mbox mbox Interfaces
+ *  @{
+ */
+
+/**
+ * @brief	Initialize a libmetal mbox.
+ * @param[in]	mbox	mbox to initialize.
+ * @param[in]	init_data	Data used to initialize mbox.
+ *				In this case the driver instance is
+ *				already set up at Zephr init.
+ * @return	0 on success. Negative value on failure.
+ */
+extern int __metal_mbox_init(metal_mbox_t *mbox,
```

Comment:
Instead of using external define weak functions in a mbox.c that return error

Reply:
ok can do!